File: src/main/java/frc/robot/subsystems/indexer/IndexerIOSim.java
package frc.robot.subsystems.indexer;

import edu.wpi.first.units.measure.Voltage;

import static edu.wpi.first.units.Units.*;

public class IndexerIOSim implements IndexerIO {

    Voltage vout = Volts.zero();


    public IndexerIOSim() {
    }

    @Override
    public void updateInputs(IndexerIOInputs input) {
        //TODO, kv is from recalc, test on bot
        input.speed = RPM.of(198.52*vout.in(Volts));

        input.motorVoltageOut = vout;

        input.motorConnected = true;
    }

    @Override
    public void setVout(Voltage vout) {
        this.vout = vout;
    }
}

File: src/main/kotlin/frc/robot/subsystems/indexer/IndexerIOSim.kt
package frc.robot.subsystems.indexer

import edu.wpi.first.units.Units.RPM
import edu.wpi.first.units.Units.Volts
import edu.wpi.first.units.measure.Voltage

class IndexerIOSim : IndexerIO {

    private var vout: Voltage = Volts.zero()

    override fun updateInputs(input: IndexerIO.IndexerIOInputs) {
        // TODO, kv is from recalc, test on bot
        input.speed = RPM.of(198.52 * vout.`in`(Volts))

        input.motorVoltageOut = vout

        input.motorConnected = true
    }

    override fun setVout(vout: Voltage) {
        this.vout = vout
    }
}
